src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -4209,5 +4209,19 @@ def test_vmss_update_automatic_repairs_with_health_extension(self, resource_grou
                  ])
 
 
+class VMCreateNSGRule(ScenarioTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_test_vm_create_nsg_rule_')
+    def test_vm_create_nsg_rule(self, resource_group):
+        self.kwargs.update({
+            'vm': 'vm1'
+        })
+
+        self.cmd('vm create -g {rg} -n {vm} --image centos --nsg-rule NONE')
+        self.cmd('network nic show -g {rg} -n {vm}VMNIC', checks=[
+            self.check('networkSecurityGroup.defaultSecurityRules', None)
```

Comment:
In my test, `networkSecurityGroup.defaultSecurityRules` is always `None`.

We should check `securityRules` from `az network nsg show` instead.

Reply:
Agree.